src/wix/WixToolset.Core/Compiler.cs

```diff
@@ -5661,13 +5685,23 @@ private void ParseNakedFileElement(XElement node, ComplexReferenceParentType par
 
                 directoryId = this.HandleSubdirectory(sourceLineNumbers, node, directoryId, subdirectory, "Directory", "Subdirectory");
 
-                this.Core.AddSymbol(new ComponentSymbol(sourceLineNumbers, fileSymbol.Id)
+                this.ParseFileNamingAttributes(node, sourcePath, directoryId, isNakedFile: true, out var id, out var name, out var shortName, out var source);
+
+                // Now that we have all the data we need to generate a good id, do
+                // so and create a file and component symbol with the right data.
+                id = id ?? this.Core.CreateIdentifier("nkf", directoryId, name, condition, win64.ToString());
+
+                var keyPath = this.ParseFileElementOtherAttributes(node, id.Id, directoryId, diskId: CompilerConstants.IntegerNotSet, id, name, shortName, source, out var _, componentGuid: "*", isNakedFile: true, fileSymbol: out var fileSymbol, assemblySymbol: out var assemblySymbol);
```

Comment:
The explicit parameter declaration looks a little funny in the out parameters that have the same names in the `out var`, like: `fileSymbol: out var fileSymbol` and `assemblySymbol: out var assemblySymbol` 

I'd remove the redundancy.

Reply:
Yeah, that's an error if you use a named argument earlier in the list. So all stay or all go. Vote?

Reply:
Oh, I didn't realize you had to use it in all the following arguments. It adds value to the ones earlier, so it's worth it. Plus, maybe someone else will learn something if they try to "clean them up" in the future. :)

Reply:
Yeah, it's not great but you gotta balance "naked `false`s" against "`out`s at the end."